### PR TITLE
Obs-only: check model_list to see if empty instead

### PR DESCRIPTION
**Pull request by lewisblake**
## Change Summary

Change `if not self.cfg.model_cfg` in experiment_processor.py to instead look at `model_list`, which should evaluate to False if empty. Note `model_list` is created for every `model_name`, but will always be empty if no models are provided.

## Related issue number

N/A, chat with @dulte 

## Checklist

* [x] Start with a draft-PR
* [x] The PR title is a good summary of the changes
* [x] PR is set to AeroTools and a tentative milestone
* [ ] Documentation reflects the changes where applicable
* [ ] Tests for the changes exist where applicable
* [x] Tests pass locally
* [x] Tests pass on CI
* [x] At least 1 reviewer is selected
* [x] Make PR ready to review

### CODE REVIEW

File: pyaerocom/helpers.py [hunk 1]
```diff
@@ -1704,18 +1726,22 @@ def get_max_period_range(periods):
...
 ) -> iris.cube.Cube:
-    startstr = f"days since {start_yr}-01-01 00:00"
+    startstr = f"{TS_TYPE_TO_FREQ_NAME[freq]} since {start_yr}-01-01 00:00"
 
     if freq not in TS_TYPE_TO_PANDAS_FREQ.keys():
         raise ValueError(f"{freq} not a recognized frequency")
 
     start_str = f"{start_yr}-01-01 00:00"
-    stop_str = f"{stop_yr}-12-31 00:00"
+    stop_str = f"{int(stop_yr)+1}-01-01 00:00"
```
Comment (thorbjoernl) [inline pyaerocom/helpers.py]: Might be wrong but `pd.date_range` is inclusive on both ends by default, so I think this includes one more element than we want? https://pandas.pydata.org/docs/reference/api/pandas.date_range.html

File: pyaerocom/aeroval/collections.py [hunk 2]
```diff
@@ -87,13 +87,18 @@ def keylist(self, name_or_pattern: str = None) -> list[str]:
         KeyError
             if no matches can be found
         """
+        # Special case where the model cfg is empty, used for obs-only
+        if self._entries.keys() == []:
```
Comment (thorbjoernl) [inline pyaerocom/aeroval/collections.py:91]: Shouldn't this be on `ModelCollection`? Now it applies to both `ObsCollection` and `ModelCollection` since it's on the base class.

File: pyaerocom/aeroval/experiment_processor.py [hunk 3]
```diff
@@ -132,19 +132,19 @@ def run(self, model_name=None, obs_name=None, var_list=None, update_interface=Tr
...
 
         obs_list = self.cfg.obs_cfg.keylist(obs_name)
-        if not self.cfg.model_cfg:
+        model_list = self.cfg.model_cfg.keylist(model_name)
+        if not model_list:
             logger.info("No model found, will make dummy model data")
             self.cfg.webdisp_opts.hide_charts = ["scatterplot"]
             self.cfg.webdisp_opts.pages = ["evaluation", "infos"]
             model_id = make_dummy_model(obs_list, self.cfg)
+            model_list.append("dummy")
```
Comment (thorbjoernl) [inline pyaerocom/aeroval/experiment_processor.py]: What is the intent here? Does this relate to the preceeding line? Maybe a clarifying comment would be useful since this does not appear related to the preceeding lines?

**Review by thorbjoernl — CHANGES_REQUESTED** [review]

File: pyaerocom/aeroval/experiment_processor.py (hunk 3 above)
Comment (dulte) [reply pyaerocom/aeroval/experiment_processor.py]: The line above adds the dummy to cfg, but not to the model_list, which is still empty. This list is used later. But I've added a comment

File: pyaerocom/aeroval/collections.py (hunk 2 above)
Comment (dulte) [reply pyaerocom/aeroval/collections.py:91]: Good observation, moved to ModelCollection

File: pyaerocom/helpers.py (hunk 1 above)
Comment (dulte) [reply pyaerocom/helpers.py]: True, this was added to bypass a bug (maybe) with reading the Cube. But I think I've fixed it, so this will be removed

File: pyaerocom/helpers.py [hunk 4]
```diff
@@ -1738,7 +1738,7 @@ def make_dummy_cube(
         raise ValueError(f"{freq} not a recognized frequency")
 
     start_str = f"{start_yr}-01-01 00:00"
-    stop_str = f"{int(stop_yr)+1}-01-01 00:00"
+    stop_str = f"{int(stop_yr)}-12-31 00:00"
```
Comment (thorbjoernl) [inline pyaerocom/helpers.py]:
```suggestion
    stop_str = f"{int(stop_yr)}-12-31 23:00"
```

**Review by thorbjoernl — APPROVED** [review]

**Review by thorbjoernl — APPROVED** [review]
Lite forslag til endring, ellers LGTM
